docs/user/rocket/generic_surface.rst

```diff
@@ -0,0 +1,422 @@
+single value, a callable function of seven arguments, or a path to a ``.csv``
+file containing the values.
+
+The coefficients are all functions of:
+
+- Angle of attack (:math:`\alpha`) in radians.
+- Side slip angle (:math:`\beta`) in radians.
+- Mach number (:math:`Ma`).
+- Reynolds number (:math:`Re`).
+- Pitch rate (:math:`q`) in radians per second.
+- Yaw rate (:math:`r`) in radians per second.
+- Roll rate (:math:`p`) in radians per second.
+
+.. math::
+   \begin{aligned}
+      C_L &= f(\alpha, \beta, Ma, Re, q, r, p) \\
+      C_Q &= f(\alpha, \beta, Ma, Re, q, r, p) \\
+      C_D &= f(\alpha, \beta, Ma, Re, q, r, p) \\
+      C_m &= f(\alpha, \beta, Ma, Re, q, r, p) \\
+      C_n &= f(\alpha, \beta, Ma, Re, q, r, p) \\
+      C_l &= f(\alpha, \beta, Ma, Re, q, r, p)
+   \end{aligned}
+
+From the coefficients, the forces and moments are calculated with
+
+.. math::
+   \begin{aligned}
+      L &= \overline{q}\cdot A_{ref}\cdot C_L \\
+      Q &= \overline{q}\cdot A_{ref}\cdot C_Q \\
+      D &= \overline{q}\cdot A_{ref}\cdot C_D \\
+      M_{m} &= \overline{q}\cdot A_{ref}\cdot L_{ref}\cdot C_m \\
+      M_{n} &= \overline{q}\cdot A_{ref}\cdot L_{ref}\cdot C_n \\
+      M_{l} &= \overline{q}\cdot A_{ref}\cdot L_{ref}\cdot C_l
+   \end{aligned}
+
+These coefficients can be defined as a callable such as:
+
+.. code-block:: python
+
+   def coefficient(alpha, beta, Ma, Re, q, r, p):
+      ...
+      return value
+
+In which any algorithm can be implemented to calculate the coefficient values.
+
+Otherwise, the coefficients can be defined as a ``.csv`` file. The file must
+contain a header with at least one of the following columns representin the
+independent variables:
+
+- ``alpha``: Angle of attack.
+- ``beta``: Side slip angle.
+- ``mach``: Mach number.
+- ``reynolds``: Reynolds number.
+- ``q``: Pitch rate.
+- ``r``: Yaw rate.
+- ``p``: Roll rate.
+
+The last column must be the coefficient value, and must contain a header, 
+though the header name can be anything.
+
+.. important::
+   Not all columns need to be present in the file, but the columns that are
+   present must be named, **and ordered**, as described above.
+
+An example of a ``.csv`` file is shown below:
+
+.. code-block:: 
+
+   "alpha", "mach", "coefficient"
+   -0.017, 0, -0.11
+   -0.017, 1, -0.127
+   -0.017, 2, -0.084
+   -0.017, 3, -0.061
+   0.0, 0, 0.0
+   0.0, 1, 0.0
+   0.0, 2, 0.0
+   0.0, 3, 0.0
+   0.017, 0, 0.11
+   0.017, 1, 0.127
+   0.017, 2, 0.084
+   0.017, 3, 0.061
+
+After the definition of the ``GenericSurface`` object, it must be added to the
+rocket's configuration:
+
+.. seealso::
+   For more information on how to add a generic surface to the rocket, see
+   :class:`rocketpy.Rocket.add_generic_surface`
+
+.. code-block:: python
+   :emphasize-lines: 5
+
+   from rocketpy import Rocket
+   rocket = Rocket(
+      ...
+   )
+   rocket.add_generic_surface(generic_surface, position=(0,0,0))
```

Comment:
Mmmmmm do we really need a `add_generic_surface` method? Why not using the very same `add_surfaces` method?